src/eredis_sentinel.erl

```diff
@@ -0,0 +1,244 @@
+                                password = Password,
+                                connect_timeout = ConnectTimeout,
+                                socket_options = SocketOptions,
+                                tls_options = TlsOptions,
+                                conn_pid = undefined,
+                                errors    = #errors{}}}.
+
+%% @doc Handling call messages
+-spec(handle_call(Request :: term(), From :: {pid(), Tag :: term()},
+                  State :: #eredis_sentinel_state{}) ->
+             {reply, Reply :: term(), NewState :: #eredis_sentinel_state{}} |
+             {reply, Reply :: term(), NewState :: #eredis_sentinel_state{}, timeout() | hibernate} |
+             {noreply, NewState :: #eredis_sentinel_state{}} |
+             {noreply, NewState :: #eredis_sentinel_state{}, timeout() | hibernate} |
+             {stop, Reason :: term(), Reply :: term(), NewState :: #eredis_sentinel_state{}} |
+             {stop, Reason :: term(), NewState :: #eredis_sentinel_state{}}).
+handle_call(get_master, _From, State) ->
+    case query_master(State#eredis_sentinel_state{errors = #errors{}}) of
+        {ok, {Host,Port}, S1} ->
+            {reply, {ok, Host,Port}, S1};
+        {error, Error, S1} ->
+            {reply, {error, Error}, S1}
+    end;
+
+handle_call(stop, _From, State) ->
+    {stop, normal, ok, State}.
+
+%% @doc Handling cast messages
+-spec(handle_cast(Request :: term(), State :: #eredis_sentinel_state{}) ->
+             {noreply, NewState :: #eredis_sentinel_state{}} |
+             {noreply, NewState :: #eredis_sentinel_state{}, timeout() | hibernate} |
+             {stop, Reason :: term(), NewState :: #eredis_sentinel_state{}}).
+handle_cast(_Request, State = #eredis_sentinel_state{}) ->
+    {noreply, State}.
+
+%% @doc Handling all non call/cast messages
+-spec(handle_info(Info :: timeout() | term(), State :: #eredis_sentinel_state{}) ->
+             {noreply, NewState :: #eredis_sentinel_state{}} |
+             {noreply, NewState :: #eredis_sentinel_state{}, timeout() | hibernate} |
+             {stop, Reason :: term(), NewState :: #eredis_sentinel_state{}}).
+%% Current sentinel connection broken
+handle_info({'EXIT', Pid, _Reason}, #eredis_sentinel_state{conn_pid = Pid} = S) ->
+    {noreply, S#eredis_sentinel_state{conn_pid = undefined}};
+
+%% Ignore late exit messages
+handle_info({'EXIT', _Pid, _Reason}, S) ->
+    {noreply, S};
+
+handle_info(_Info, State) ->
+    {noreply, State}.
+
+%% @doc This function is called by a gen_server when it is about to
+%% terminate. It should be the opposite of Module:init/1 and do any
+%% necessary cleaning up. When it returns, the gen_server terminates
+%% with Reason. The return value is ignored.
+-spec(terminate(Reason :: (normal | shutdown | {shutdown, term()} | term()),
+                State :: #eredis_sentinel_state{}) -> term()).
+terminate(_Reason, _State = #eredis_sentinel_state{conn_pid = Pid}) ->
+    eredis:stop(Pid),
```

Comment:
Do we need to handle the case `conn_pid = undefined` here?

Reply:
If we want to cover these cases with test cases, we can find the linked processes using `process_info(Pid, links)` and kill or call `eredis:stop` on one of them. Just an idea.